src/client/datascience/jupyter/kernels/kernelSelector.ts

```diff
@@ -28,19 +21,16 @@ import { Uri } from 'vscode';
  */
 @injectable()
 export class KernelSelector {
-    private readonly kernelSelectionsByResource = new Map<Uri, Promise<KernelConnectionMetadata | undefined>>();
     constructor(
-        @inject(KernelSelectionProvider) private readonly selectionProvider: KernelSelectionProvider,
         @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
-        @inject(IConfigurationService) private configService: IConfigurationService,
-        @inject(InterpreterPackages) private readonly interpreterPackages: InterpreterPackages
-    ) {}
+        @inject(ICommandManager) private readonly commandManager: ICommandManager,
+        @inject(IServiceContainer) protected serviceContainer: IServiceContainer // @inject(IInteractiveWindowProvider) private readonly interactiveWindowProvider: IInteractiveWindowProvider
```

Comment:
This line looks off here.

Reply:
Looks like a cyclic reference,
when we refactor and remove these layers this should all go away, my Pr is merely a hacky solution